### PR TITLE
otpe

### PR DESCRIPTION
**Pull request by gheorghestrimtu**
**Change:**
Broke `NewK8sEnvironment` into two parts. One part will connect to the cluster, the second part will deploy specs. 

**Motivation**
OTPE needs 2 urls that point to configs that contain lists of contract addresses and node addresses. In order to provide this contact addresses the contracts have to be deployed during the test. This is done through the `BlockchainClient`. The constructor of `BlockchainClient` needs an `Environment` variable that is returned by `NewK8sEnvironment`. With the previous setup you cannot invoke the `BlockchainClient` between deployments of different manifests.

**Change:**
Added mockserver helm chart
**Motivation:**
otpe needs to read configs from 2 endpoints that in prod come from weiwatchers, this endpoints need to be mocked with test info

### CODE REVIEW

File: suite/alerts/alerts_test.go [hunk 1]
```diff
@@ -1,54 +1,24 @@
...
-			Expect(err).ShouldNot(HaveOccurred())
-			fmt.Println(chainlinkNodes[0].URL())
-
-			adapter, err = environment.GetExternalAdapter(suiteSetup.Env)
-			Expect(err).ShouldNot(HaveOccurred())
-			fmt.Println(adapter.ClusterURL())
-		})
-	})
-
+var _ = FDescribe("Alerts suite", func() {
```
Comment (skudasov) [inline suite/alerts/alerts_test.go:10]: Please, remove focus.

File: suite/alerts/alerts_test.go [hunk 2]
```diff
@@ -1,54 +1,24 @@
...
 	Describe("Alerts", func() {
 		It("Test 1", func() {
```
Comment (skudasov) [inline suite/alerts/alerts_test.go:12]: Can we have more explicit naming for that test? Is it testing just the deployment with OPTE?

File: environment/templates/explorer-deployment.yml [hunk 3]
```diff
@@ -60,16 +58,7 @@ spec:
...
-          volumeMounts:
-            - name: kafka-tls
-              mountPath: /etc/ssl/certs/kafka-0.keystore.pem
-              subPath: kafka-0.keystore.pem
-              readOnly: false
-            - name: kafka-tls
-              mountPath: /etc/ssl/certs/kafka.truststore.pem
-              subPath: kafka.truststore.pem
-              readOnly: false
+          image: 'studentcuza/explorer:choose-security-protocol'
```
Comment (skudasov) [inline environment/templates/explorer-deployment.yml:61]: Maybe we can add explorer image as a var in our main config, repo/tag?

File: environment/templates/otpe-deployment.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+              value: ":1337"
+            - name: WEIWATCHERS_NODES_URL
+              value: {{ .Values.MockserverDependencyGroup.mockserver.nodesURL }}
+            - name: WEIWATCHERS_CHAINMONITOR_URL
+              value: {{ .Values.MockserverDependencyGroup.mockserver.contractsURL }}
+            - name: CHAIN_TYPE
+              value: eth
+            - name: LOG_LEVEL
+              value: debug
+          image: '804282218731.dkr.ecr.us-west-2.amazonaws.com/ocr-telemetry-prometheus-exporter:2496ee62ed6c81898ddcc42d40d3be1f9c7ac549'
```
Comment (skudasov) [inline environment/templates/otpe-deployment.yml:38]: And here too, image/tag would be nice in config.

**Review by kalverra — CHANGES_REQUESTED** [review]

File: config.yml [hunk 5]
```diff
@@ -38,6 +38,12 @@ apps:
         limits:
             memory: 96Mi
             cpu: 50m
+    otpe:
```
[the remark this answers is not in the document]
Comment (gheorghestrimtu) [reply config.yml:41]: Sergey suggested this change

File: environment/k8s_environment.go [hunk 6]
```diff
@@ -123,26 +126,39 @@ func NewK8sEnvironment(
 	}
 	env.chaos = cc
 
+	return env, nil
+}
+
+// DeploySpecs deploys a all the specs in the array
```
Comment (kalverra) [inline environment/k8s_environment.go:132]: _deploys all the specs in the array_

`the array` is not exactly clear as to what we're deploying from. Maybe a better version would be something like 

> deploys all specs in the provided environment init function

>

**Review by kalverra — CHANGES_REQUESTED** [review]

File: config.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (kalverra) [reply config.yml:41]: Ah yes, my bad, missed the section it was in, carry on.

**Review by kalverra — APPROVED** [review]

**Pull request merged**